imap/jmap_mail.c

```diff
@@ -13236,6 +13236,63 @@ static int msgimport_checkacl_cb(const mbentry_t *mbentry, void *xrock)
     return 0;
 }
 
+static time_t _email_import_parse_received_at(const char *blob, size_t blob_len)
+{
+    message_t *msg = message_new_from_data(blob, blob_len);
+    struct buf buf = BUF_INITIALIZER;
+    time_t received_at = 0;
+    enum message_format format =  MESSAGE_DECODED | MESSAGE_TRIM;
+
+    if (!msg)
+        goto done;
+
+    if (!message_get_field(msg, "X-Deliveredinternaldate", format, &buf)) {
+        buf_appendcstr(&buf, "\r\n");
+        const char *hdr = buf_cstring(&buf);
+        char *val = NULL;
+        message_parse_string(hdr, &val);
+        if (time_from_rfc822(val, &received_at) < 0)
+            received_at = 0;
+        xzfree(val);
+    }
+
+    if (received_at)
+        goto done;
+
```

Comment:
Just a nit, but rather than using the goto above, the conditional below code be

if (!received_at && !message_get_field(...

Reply:
It figured it's more obvious that we return early this way, but I can change that if you prefer.

Reply:
Its not a huge deal.  Or maybe put that check inside the if {} after the xzfree()

Reply:
I now rewrote this to use `if (!received_at && !message_get_field(..`